R/render-table.R

```diff
@@ -3,33 +3,104 @@
+#'   the corresponding Bootstrap table format to the output table.
+#' @param width Table width. Must be a valid CSS unit (like "100%", 400px",
+#'   "auto") or a number, which will be coerced to a string and
+#'    have "px" appended.
+#' @param align A string that specifies the column alignment. If equal to
+#'   \code{'l'}, \code{'c'} or \code{'r'}, then all columns will be,
+#'   respectively, left-, center- or right-aligned. Otherwise, \code{align}
+#'   must have the same number of characters as the resulting table (if
+#'   \code{rownames = TRUE}, this will be equal to \code{ncol()+1}), with
+#'   the \emph{i}-th character specifying the alignment for the
+#'   \emph{i}-th column (again, only \code{'l'}, \code{'c'} or
+#'   \code{'r'} are permitted). If \code{NULL}, then all numeric/integer
+#'   columns (including the row names, if they are numbers) will be
+#'   right-aligned and everything else will be left-aligned.
+#' @param rownames,colnames Logicals: include rownames? include colnames
+#'   (column headers)?
+#' @param digits An integer specifying the number the decimal places for
+#'   the numeric columns (this will not apply to columns with an integer
+#'   class). If \code{digits} is set to a negative value, then the numeric
+#'   columns will be displayed in scientific format with a precision of
+#'   \code{abs(digits)} digits.
+#' @param na The string to use in the table cells whose values are missing
+#'   (i.e. they either evaluate to \code{NA} or \code{NaN}).
+#' @param ... Arguments to be passed through to \code{\link[xtable]{xtable}}
+#'   and \code{\link[xtable]{print.xtable}}.
 #' @param env The environment in which to evaluate \code{expr}.
-#' @param quoted Is \code{expr} a quoted expression (with \code{quote()})? This
-#'   is useful if you want to save an expression in a variable.
+#' @param quoted Is \code{expr} a quoted expression (with \code{quote()})?
+#'   This is useful if you want to save an expression in a variable.
 #' @param func A function that returns an R object that can be used with
 #'   \code{\link[xtable]{xtable}} (deprecated; use \code{expr} instead).
 #'
 #' @export
-renderTable <- function(expr, ..., env=parent.frame(), quoted=FALSE, func=NULL) {
+renderTable <- function(expr, striped=FALSE, condensed=TRUE,
+                        hover=FALSE, bordered=FALSE,
+                        width="auto", align=NULL,
+                        rownames=FALSE, colnames=TRUE,
+                        digits=NULL, na="NA", ...,
+                        env=parent.frame(), quoted=FALSE, func=NULL) {
   if (!is.null(func)) {
     shinyDeprecated(msg="renderTable: argument 'func' is deprecated. Please use 'expr' instead.")
   } else {
     installExprFunction(expr, "func", env, quoted)
   }
 
+  # A small helper function to create a wrapper for an argument that was
+  # passed to renderTable()
+  createWrapper <- function(arg) {
+    if (is.function(arg)) wrapper <- reactive({ arg() })
```

Comment:
I'm not sure why we did that in renderPlot. On the pro side it will cache the value (actually not totally sure if that's good or bad). On the bad side it will add noise to the reactive graph.